TransactionService/docker_context/handlers/BaseLogicHandler.py

```diff
@@ -0,0 +1,117 @@
+import pickle
+from datetime import datetime
+
+class BaseLogicHandler:
+
+    def __init__(self, LegacyStockServerHandler, redis, RedisHandler, audit, client, ip, port):
+        self.LegacyStockServer = LegacyStockServerHandler
+        self.redis = redis
+        self.RedisHandler = RedisHandler
+        self.audit = audit
+        self.client = client
+        self.dbmURL = f'HTTP://{ip}:{port}'
+
+    async def getRequest(self, url, params=None):
+        async with self.client.get(url, params=params) as resp:
+            js = await resp.json()
+            return js
+
+    async def postRequest(self, url, data):
+        async with self.client.post(url, json=data) as resp:
+            js = await resp.json()
+            return js
+
+    async def addFunds(self, trans_num, user_id, amount):
+        data = {'user_id': user_id, 'funds': float(amount)}
+        result = await self.postRequest(self.dbmURL+"/funds/add_funds", data)
+        return result
+
+    async def getQuote(self, trans_num, user_id, stock_id):
+        result = await self.LegacyStockServer.getQuote(trans_num, user_id, stock_id)
+        return result
+
+    async def BuyStock(self, trans_num, user_id, stock_symbol, amount):
+        if self.redis.exists(user_id):
+            user = pickle.loads(self.redis.get(user_id))
+            price = float(await self.LegacyStockServer.getQuote(trans_num, user_id, stock_symbol))
+            amountStock = float(amount)/price
+            totalValue = amountStock*price
+            if float(user["account_balance"]) >= float(amount):
+                dictionary = {"user_id": user_id, "stock_id": stock_symbol, "amount": totalValue,
+                              "amount_of_stock": amountStock, "time": datetime.now()}
+                self.redis.set(user_id + "_BUY", pickle.dumps(dictionary))
+                return {"statusMessage": "Buy Created"}
+            else:
+                return {'errorMessage': 'Specified user does not have sufficient funds'}
+        else:
+            return {'errorMessage': 'Specified user does not exist'}
+
+    async def commitBuy(self, trans_num, user_id):
+        if self.redis.exists(user_id + "_BUY"):
+            buyObj = pickle.loads(self.redis.get(user_id + "_BUY"))
+            now = datetime.now()
+            timeDiff = (now - buyObj["time"]).total_seconds()
+            if timeDiff <= 60:
+                data = {"user_id": user_id, "funds": buyObj["amount"],
+                        "stock_symbol": buyObj["stock_id"], "stock_amount": buyObj["amount_of_stock"]}
+                result = await self.postRequest(self.dbmURL+'/stocks/buy_stocks', data)
+                self.redis.delete(user_id + "_BUY")

```

Comment:
this is what I meant before with the redis connection pool being in the redis handler. As it is we have redis operations being done in different areas. What do you think?

Reply:
We can talk about this in the meeting in a bit, and I can rewrite if necessary.